src/ScreenCover/index.js

```diff
@@ -1,23 +1,61 @@
-import React, { Fragment } from 'react'
+import React, { Fragment, PureComponent } from 'react'
 import propTypes from 'prop-types'
 import css from 'styled-jsx/css'
 import cx from 'classnames'
 
-export const ScreenCover = ({ onClick }) => (
-    <div className="screen-cover" onClick={onClick}>
-        <style jsx>{`
-            div {
-                background: rgba(200, 200, 200, 0.6);
-                height: 100%;
-                left: 0;
-                position: absolute;
-                top: 0;
-                width: 100%;
-            }
-        `}</style>
-    </div>
-)
+class ScreenCover extends PureComponent {
+    elContainer = React.createRef()
+
+    componentDidMount() {
+        document.addEventListener('click', this.onDocClick)
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('click', this.onDocClick)
+    }
+
+    onDocClick = evt => {
+        console.log(this.elContainer, evt)
+        if (
+            this.elContainer.current &&
+            this.elContainer.current === evt.target
+        ) {
+            this.props.onClick()
+        }
+    }
+
+    render() {
+        return (
+            <div className="screen-cover" ref={this.elContainer}>
+                <div className="children">{this.props.children}</div>
+
+                <style jsx>{`
+                    .screen-cover {
+                        display: flex;
+                        align-items: center;
+                        justify-content: center;
+
+                        position: fixed;
```

Comment:
The container needs a z-index methinks. Perhaps the children don't... But the `.screen-cover` does.

Reply:
6f6b48b